packages/shared/src/components/post/write/SubmitExternalLink.tsx

```diff
@@ -0,0 +1,155 @@
+import React, { FormEventHandler, ReactElement, useState } from 'react';
+import classNames from 'classnames';
+import { ExternalLinkPreview, ReadHistoryPost } from '../../../graphql/posts';
+import { TextField } from '../../fields/TextField';
+import { Loader } from '../../Loader';
+import LinkIcon from '../../icons/Link';
+import { ClickableText } from '../../buttons/ClickableText';
+import { LazyModal } from '../../modals/common/types';
+import { useLazyModal } from '../../../hooks/useLazyModal';
+import useDebounce from '../../../hooks/useDebounce';
+import { isValidHttpUrl } from '../../../lib/links';
+import { Image } from '../../image/Image';
+import { Button } from '../../buttons/Button';
+import OpenLinkIcon from '../../icons/OpenLink';
+import { SourceAvatar } from '../../profile/source';
+import { ElementPlaceholder } from '../../ElementPlaceholder';
+import useMedia from '../../../hooks/useMedia';
+import { tablet } from '../../../styles/media';
+
+interface SubmitExternalLinkProps {
+  preview: ExternalLinkPreview;
+  isLoadingPreview: boolean;
+  getLinkPreview: (value: string) => void;
+  onSelectedHistory: (post: ReadHistoryPost) => void;
+}
+
+export function SubmitExternalLink({
+  onSelectedHistory,
+  isLoadingPreview,
+  getLinkPreview,
+  preview,
+}: SubmitExternalLinkProps): ReactElement {
+  const isMobile = !useMedia([tablet.replace('@media ', '')], [true], false);
+  const { openModal } = useLazyModal();
+  const [url, setUrl] = useState<string>(undefined);
+  const shouldShorten = url !== undefined || isMobile;
+  const label = `Enter URL${shouldShorten ? '' : ' / Choose from'}`;
+  const [checkUrl] = useDebounce((value: string) => {
+    if (!isValidHttpUrl(value) || value === preview?.url) return null;
+
+    return getLinkPreview(value);
+  }, 1000);
+
+  const onInput: FormEventHandler<HTMLInputElement> = (e) => {
+    const { value } = e.currentTarget;
+    setUrl(value);
+    checkUrl(value);
+  };
+
+  const imageClass = 'w-28 h-16 rounded-12 bg-theme-label-disabled';
+  const link = preview?.url ?? preview?.permalink ?? url;
+
+  if (preview || isLoadingPreview) {
```

Comment:
I think we should create two util components for the branching here. Will this component more readable